### DISCUSSION TITLE
How to route dynamic files?

### DISCUSSION THREAD

**Discussion by bsgies** [vinissimus/next-translate#824] [Q&A]
Hi,

As shown in the screenshot below. You can see I have a folder password/reset/[hash.js].

This route keeps changing because every user has a different HASH.

I have also imported:

import useTranslation from "next-translate/useTranslation";

and declared

let { t } = useTranslation("hash");

How do I define route in i18n.json?

<img width="986" alt="image" src="https://user-images.githubusercontent.com/32961638/161039989-4bed8200-3bc9-4b2a-b550-2afb532a4fc4.png">

**Comment by aralroca** [comment]
maybe I misunderstood you, you mean how to put it inside `pages`? This would be enough:

i18n.js
```js
{
 ...restOfConfig,
  "pages": {
     "/password/reset/[hash]": ["hash"],
  }
}
```

**Reply by abednaseri** [reply]
thank you! was looking for that.

**Comment by abednaseri** [comment]
Nice answer, do you know how I can do a similar thing with multiple pages inside a folder? So I have the following structure with my pages.

```
/pages/legal/a.tsx
/pages/legal/b.tsx
/pages/legal/c.tsx
```

And would like to have only one json file called legal, to use it for all those 3 pages.

I tried with a star * in my config but didn't work.

```
pages: {
    '*': ['common'],
    '/legal/*': ['legal'],
  },
```

**Reply by aralroca** [reply]
Yes, with regex! You can see an example here: https://github.com/aralroca/next-translate/blob/321016c13fb5678e31e79336facde0398c58d96d/examples/complex/i18n.js#L10

You need to add the prefix `rgx:`
